src/Sarif/Core/PropertyBagHolder.cs

```diff
@@ -63,17 +61,19 @@ public string GetProperty(string propertyName)
                         propertyName));
             }
 
+            // Avoid a crash if the string-valued property we are attempting to retrieve has
+            // a null value in the JSON file.
+            if (Properties[propertyName] == null) { return null; }
+
             if (!Properties[propertyName].IsString)
             {
                 throw new InvalidOperationException(SdkResources.CallGenericGetProperty);
             }
 
             string value = Properties[propertyName].SerializedValue;
 
-            // Remove the quotes around the serialized value ("x" => x) -- unless it's null.
-            return value.Equals(NullValue, StringComparison.Ordinal)
-                ? null
```

Comment:
>null [](start = 18, length = 4)

This null check is no longer needed because we simplified the way that `SetProperty` deals with `null` values (`SetProperty("aProp", null)`). Rather than creating a `SerializedPropertyInfo` object with `SerializedValue` set to `null` and placing that object into `Properties["aProp"]`, we simply set `Properties["aProp"]` to `null`.

This is consistent with what the `PropertyBagConverter` does when it encounters a `null` valued property. Before this change, a `null` valued property read from the log file had a different internal representation from a property you created programatically with `SetProperty("aProp", null)`.

This is one of those explanatory comments that doesn't have a natural place in the source code. I'm explaining why some code _no longer appears_. But, come to think of it, I can add a comment on the `null` path in `SetProperty`, so I'll do that.